slick.grid.js

```diff
@@ -558,69 +572,67 @@ if (typeof Slick === "undefined") {
         resizeCanvas();
         bindAncestorScrollEvents();
 
-        $container
-            .on("resize.slickgrid", resizeCanvas);
-        $viewport
-            .on("scroll", handleScroll);
+        _container.addEventListener("resize.slickgrid", resizeCanvas);
+        _viewport.forEach((view) => {
+          view.addEventListener("scroll", handleScroll);
+        });
 
         if (options.enableMouseWheelScrollHandler) {
-          $viewport.each(function (_idx, viewportElm) {
+          _viewport.forEach((view) => {
             slickMouseWheelInstances.push(Slick.MouseWheel({
-              element: viewportElm,
+              element: view,
               onMouseWheel: handleMouseWheel
             }));
           });
         }
 
-        $headerScroller
+        _headerScroller.forEach((el) => {
             //.on("scroll", handleHeaderScroll)
-            .on("contextmenu", handleHeaderContextMenu)
-            .on("click", handleHeaderClick)
-            .on("mouseenter", ".slick-header-column", handleHeaderMouseEnter)
-            .on("mouseleave", ".slick-header-column", handleHeaderMouseLeave);
-        $headerRowScroller
-            .on("scroll", handleHeaderRowScroll);
+            el.addEventListener("contextmenu", handleHeaderContextMenu);
+            el.addEventListener("click", handleHeaderClick);
+        });
+
+        _headerRowScroller.forEach((scroller) => {
+           scroller.addEventListener("scroll", handleHeaderRowScroll);
+        });
 
         if (options.showHeaderRow) {
-          $headerRow
-            .on("mouseenter", ".slick-headerrow-column", handleHeaderRowMouseEnter)
-            .on("mouseleave", ".slick-headerrow-column", handleHeaderRowMouseLeave);
+          _headerRows.forEach((row) => {
+            //row.addEventListener("mouseover", handleHeaderRowMouseOver);
```

Comment:
what's the reason that these were commented out and removed? That is probably why you had to fix the Cypress tests on the CustomTooltip Plugin, isn't it?

Reply:
Well, I'm not a web guy, therefore I always learn new stuff whenever I do some javascript :) And I learned that javascript has two different ways to trigger a hover event: mouseover/out and mouseenter/leave. And I actually don't know what jQuery is doing under the hood, therefore I was playing around a little bit and probably forgot to remove this old code.

Anyway, the canvas uses mouseover/mouseout and all the other elements use mouseenter/mouseleave.

The problem with your cypress test was, that you mixed mouseover/mouseleave (or the other way around, I don't know anymore). While this works with the secret jQuery sauce and all the events where called, it didn't work in pure javascript. That made me mad and took some time to debug :)